src/lib/Libdb/db_postgres_attr.c

```diff
@@ -281,21 +274,38 @@ attrlist_to_dbarray_ex(char **raw_array, pbs_db_attr_list_t *attr_list, int keys
 	val = (struct str_data *)((char *) array + sizeof(struct pg_array));
 
 	for (pal = (svrattrl *)GET_NEXT(attr_list->attrs); pal != NULL; pal = (svrattrl *)GET_NEXT(pal->al_link)) {
-		sprintf(val->str, "%s.%s", pal->al_atopl.name, ((pal->al_atopl.resource == NULL) ? "" : pal->al_atopl.resource));
-		val->len = htonl(strlen(val->str));
-
-		val = (struct str_data *)(val->str + ntohl(val->len)); /* point to end */
+		spc_avl =  len - ((void *) val - (void *) array);
```

Comment:
shall we typecast to (char *) instead of (void *).
arithmetic on void * is allowed only as an extension in gcc.
https://gcc.gnu.org/onlinedocs/gcc-4.5.0/gcc/Pointer-Arith.html#Pointer-Arith